Securitycenter notification samples

cluebyte opened the pull request:
Hi! 

This is to add Notifications samples to the Security Center docs in preparation for GA.

I have a question about the expectation / process for unit tests. I didn't see much guidance in the internal documentation on how to do this. What I've done is written integration tests that actually hit production when run for each of the CRUD methods. Do these tests run as part of a CI build, or is this entirely user driven?

If the former, I think we need to add the service account(s) that will be used to authenticate when hitting the Security Center API for the project .org specified in the tests. If it's the latter, users will need to tweak the org_id to be whatever org they're using, and make sure to follow instructions on authenticating their application default credentials.

Let me know how to proceed on tests.

dazuma commented on securitycenter/Gemfile:
```diff
@@ -0,0 +1,9 @@
+group :test do
+  gem "rspec"
+  gem "rspec-retry"
+  gem "google-gax"
```
Please order these gems alphabetically (`google-gax` should come first). This is triggering the linter and causing CI to fail.

cluebyte replied:
I have reordered this now, thanks for pointing this out!

cluebyte commented:
I've made sure to include the require statements in each snippet, and also ran rubocop again - no complaints!

Strangely it didn't complain about the Gemfile the first time - perhaps rubocop doesn't lint that?

TheRoyalTnetennba commented on securitycenter/notification.rb:
```diff
@@ -0,0 +1,152 @@
+end
+# [END scc_create_notification_config]
+
+# [START scc_update_notification_config]
```
Can you move the region tags to inside the methods?

cluebyte replied:
I've migrated the region tags inside the methods now. PTAL!

TheRoyalTnetennba commented:
LGTM pending CI

cluebyte commented:
Can we run the CI tests again? I changed the bundler version to 1.17.2, and also updated our test topic permissions so the kokoro service account users have permissions to publish to that topic.

dazuma commented:
I enabled the API on all 11 ruby-samples test projects. Rerunning tests.

dazuma commented:
Okay there remain permissions errors that I cannot solve, while running the CI. See https://source.cloud.google.com/results/invocations/277eeaae-a9b0-4aa4-8ac3-1c7161ef0cd1/targets/cloud-devrel%2Fruby%2Fruby-docs-samples%2Fpresubmit%2Fruby-2.5/log
